atom/browser/net/url_request_context_getter.cc

```diff
@@ -341,22 +342,37 @@ net::URLRequestContext* URLRequestContextGetter::GetURLRequestContext() {
       host_resolver = std::move(remapped_resolver);
     }
 
-    net::HttpAuthPreferences auth_preferences;
+    std::vector<std::string> supported_schemes = {
+      net::kBasicAuthScheme,
+      net::kDigestAuthScheme,
+#if defined(USE_KERBEROS)
+      net::kNegotiateAuthScheme,
```

Comment:
Interesting, is this a build flag?

Reply:
Yup https://cs.chromium.org/chromium/src/net/features.gni?l=24

Reply:
[negotiate is disabled](https://github.com/electron/electron/issues/14411#issuecomment-448786510) in my build of this branch. 

- proxy auth (where I can't disable ntlm) is doing ntlm instead of negotiate
- regular auth (where only negotiate has been allowed) is failing with a 401, without a login event

Is seems that USE_KERBEROS is not defined. Does something beyond [normal](https://github.com/electron/electron/blob/3-1-x/docs/development/build-instructions-windows.md) build directions need to be done to ensure it is?

Reply:
@bughit thanks for testing, I see the issue. For 3.x libcc builds with the flag and defines it, its not carried over to electron build. Can you try removing it and see if it solves the issue ? Thanks!

Reply:
@bughit thanks for testing! I see the issue. For 3.x, libcc does build with negotiate support and defines `USE_KERBEROS`. Just that its not carried over to the electron build https://chromium.googlesource.com/chromium/src/+/66.0.3359.181/net/BUILD.gn#70, can you test with that the condition removed in this patch. Thanks!

Reply:
I am seeing the following problems with the x86 debug build (with USE_KERBEROS removed)

- the [original](https://github.com/electron/electron/issues/14411#issue-356133261) consistent crash on exit is gone but there's an intermittent check failure (even unproxied runs) also on exit.
   ```
   [2584:1220/171031.977:FATAL:ChromeClientImpl.cpp(925)] Check failed: !widget || widget->GetLayerTreeView() || !has_event_handlers.
   Backtrace:
           base::NullableString16::NullableString16 [0x69FE7627+329243]
           base::NullableString16::NullableString16 [0x69FE7731+329509]
           base::NullableString16::NullableString16 [0x6A06F39E+885650]
           blink::ElementShadow::AddShadowRoot [0x61C6FAC0+24214184]
           blink::ElementShadow::AddShadowRoot [0x612E3F2A+14204690]
           blink::ElementShadow::AddShadowRoot [0x61303A17+14334463]
           blink::ElementShadow::AddShadowRoot [0x61303ABA+14334626]
           blink::ElementShadow::AddShadowRoot [0x612C045D+14058565]
           blink::ElementShadow::AddShadowRoot [0x612EFBBA+14252962]
           blink::ElementShadow::AddShadowRoot [0x60F574BC+10482852]
           blink::ElementShadow::AddShadowRoot [0x60F691D7+10555839]
           blink::ElementShadow::AddShadowRoot [0x60F69114+10555644]
           blink::ElementShadow::AddShadowRoot [0x622566AC+30402196]
           blink::ElementShadow::AddShadowRoot [0x622567A0+30402440]
           v8::internal::CodeStubAssembler::IsSetSmi [0x6A871663+605917]
           v8::internal::CodeStubAssembler::IsSetSmi [0x6A8084EB+175461]
           v8::internal::CodeStubAssembler::IsSetSmi [0x6A86EF18+595858]
           v8::internal::CodeStubAssembler::IsSetSmi [0x6A85AF8E+514056]
   ```
- the [original](https://github.com/electron/electron/issues/14411#issuecomment-444590623) consistent crash before proxy login event is gone, but there is an intermittent check failure at the same point in time
   ```
   [2018-12-20T22:59:32.736Z] did-start-loading
   
   [2018-12-20T22:59:32.740Z] did-start-navigation: https://www.google.com/
   
   [2040:1220/175932.744:FATAL:process_metrics_win.cc(332)] Check failed: result. : The handle is invalid. (0x6)
   Backtrace:
           base::NullableString16::NullableString16 [0x69FE7627+329243]
           base::NullableString16::NullableString16 [0x69FE7731+329509]
           base::NullableString16::NullableString16 [0x6A06F39E+885650]
           base::NullableString16::NullableString16 [0x6A07056C+890208]
           base::NullableString16::NullableString16 [0x6A0E4FEE+1368034]
           base::NullableString16::NullableString16 [0x6A0F9F95+1453961]
           atom::api::App::ChildProcessLaunched [0x0113BDF0+80] (c:\users\alex\src\electron\atom\browser\api\atom_api_app.cc:813)
           atom::api::App::BrowserChildProcessLaunchedAndConnected [0x0113B166+38] (c:\users\alex\src\electron\atom\browser\api\atom_api_app.cc:780)
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x588447B4+22825143]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x58784DC3+22040262]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x5878EFE4+22081767]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x58795ED4+22110167]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x58864696+22955929]
           base::NullableString16::NullableString16 [0x6A017C05+527353]
           base::NullableString16::NullableString16 [0x6A018815+530441]
           base::NullableString16::NullableString16 [0x6A0960D8+1044684]
           base::NullableString16::NullableString16 [0x6A096310+1045252]
           base::NullableString16::NullableString16 [0x6A083988+969084]
           base::NullableString16::NullableString16 [0x6A0859C3+977335]
           base::NullableString16::NullableString16 [0x6A085553+976199]
           base::NullableString16::NullableString16 [0x6A095E7B+1044079]
           base::NullableString16::NullableString16 [0x6A095C6A+1043550]
           base::NullableString16::NullableString16 [0x6A17A204+1978872]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x588408D3+22809046]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x58864344+22955079]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x588639B5+22952632]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x5881AE3A+22654781]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x5A885687+56645514]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x5A88556A+56645229]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x5A881394+56628375]
           service_manager::MergeManifestWithOverlay [0x5610F0C0+187689]
           content::AudioRendererHost::OnRequestDeviceAuthorization [0x5A8830A9+56635820]
           wWinMain [0x010D7940+848] (c:\users\alex\src\electron\atom\app\atom_main.cc:153)
           invoke_main [0x01723B6E+30] (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:123)
           __scrt_common_main_seh [0x01723A60+336] (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283)
           __scrt_common_main [0x0172390D+13] (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:326)
           wWinMainCRTStartup [0x01723BC8+8] (f:\dd\vctools\crt\vcstartup\src\startup\exe_wwinmain.cpp:17)
           BaseThreadInitThunk [0x753F343D+18]
           RtlInitializeExceptionChain [0x777A9802+99]
           RtlInitializeExceptionChain [0x777A97D5+54]
   ```
- and there is a new major regression. If the navigation resource itself requires negotiate authentication, after creds for it are supplied in the login callback, electron hangs (navigation never completes)
  ```
  [2018-12-20T23:14:55.778Z] did-start-loading
  
  [2018-12-20T23:14:55.792Z] did-start-navigation: https://url_requiring_negotiate/
  
  [2018-12-20T23:14:56.884Z] webContents.login:
  { headers:
     { Accept:
        'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
       'Upgrade-Insecure-Requests': '1',
       'User-Agent':
        'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Electron/3.1.0-beta.4 Safari/537.36' },
    method: 'GET',
    referrer: '',
    url: 'https://url_requiring_negotiate/',
    isProxy: true,
    scheme: 'negotiate',
    host: 'proxy_host',
    port: 8080,
    realm: '' }
  ,
  
  [2018-12-20T23:14:57.017Z] webContents.login:
  { headers:
     { Accept:
        'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
       'Upgrade-Insecure-Requests': '1',
       'User-Agent':
        'Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Electron/3.1.0-beta.4 Safari/537.36' },
    method: 'GET',
    referrer: '',
    url: 'https://url_requiring_negotiate/',
    isProxy: false,
    scheme: 'negotiate',
    host: 'url_requiring_negotiate',
    port: 443,
    realm: '' }
  ,
  ```
  - nothing further is logged, the processes remain, idling
  - AFAICT the hang is before the kerbros ticket request
  - in 2.0 this works fine
  - in 3.0 there's no hang, but the familiar crash on exit
  - 3.1 with this patch and 4.0 both hang

Do you want separate bugs for any of these?

Reply:
@bughit please go ahead and create separate issue for the request hang. The other crashes are not related to this PR, but do create separate issues for them if there is a reliable repro. Thanks!

Reply:
@deepak1556 How would the failed checks manifest in official release builds, also as crashes?

Reply:
@bughit those are DCHECK failures, they wouldn't appear in release builds. Also you reported by testing on `x86 debug build`